src/librustc/mir/interpret/allocation.rs

```diff
@@ -117,9 +119,8 @@ impl<Tag> Allocation<Tag> {
     }
 
     pub fn undef(size: Size, align: Align) -> Self {
-        assert_eq!(size.bytes() as usize as u64, size.bytes());
         Allocation {
-            bytes: vec![0; size.bytes() as usize],
```

Comment:
maybe give `Size` a method `bytes_usize` which does this check for you so it isn't spread around everywhere. Make it `#[track_caller]` though

Reply:
I added `bytes_usize`, but was worried `track_caller` would be bad for performance.